src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -373,6 +373,7 @@ export interface ITerminalConfigurationService {
 	setPanelContainer(panelContainer: HTMLElement): void;
 	configFontIsMonospace(): boolean;
 	getFont(w: Window, xtermCore?: IXtermCore, excludeDimensions?: boolean): ITerminalFont;
+	updateHasCwdDetectionCapability(hasCwdDetection: boolean): void;
```

Comment:
Having anything be able to set a value like this is bad, if you were looking at `setPanelContainer` as something similar that's a bit hacky 😅 

Probably just putting this stuff into terminalInstance is the right approach.